--- example1-orig.rs
+++ example1-patched.rs
@@ -9,7 +9,7 @@
 #[derive(Debug, Clone, Default, aldrin::Serialize, aldrin::Deserialize, aldrin::AsSerializeArg)]
 #[aldrin(schema = "example1")]
 #[non_exhaustive]
-pub struct MyStructRenamed {
+pub struct MyStructRenamedAgain {
     #[aldrin(id = 1, optional)]
     pub field1: Option<u32>,
 
@@ -17,18 +17,18 @@
     pub field2: Option<u32>,
 }
 
-impl MyStructRenamed {
+impl MyStructRenamedAgain {
     pub fn new() -> Self {
         Default::default()
     }
 
-    pub fn builder() -> MyStructRenamedBuilder {
-        MyStructRenamedBuilder::new()
+    pub fn builder() -> MyStructRenamedAgainBuilder {
+        MyStructRenamedAgainBuilder::new()
     }
 }
 
 #[derive(Debug, Clone, Default)]
-pub struct MyStructRenamedBuilder {
+pub struct MyStructRenamedAgainBuilder {
     #[doc(hidden)]
     field1: Option<u32>,
 
@@ -37,7 +37,7 @@
 
 }
 
-impl MyStructRenamedBuilder {
+impl MyStructRenamedAgainBuilder {
     pub fn new() -> Self {
         Default::default()
     }
@@ -52,8 +52,8 @@
         self
     }
 
-    pub fn build(self) -> MyStructRenamed {
-        MyStructRenamed {
+    pub fn build(self) -> MyStructRenamedAgain {
+        MyStructRenamedAgain {
             field1: self.field1,
             field2: self.field2,
         }
