--- example1-orig.rs
+++ example1-patched.rs
@@ -8,22 +8,22 @@
 
 #[derive(Debug, Clone, Default)]
 #[non_exhaustive]
-pub struct MyStruct {
+pub struct MyStructRenamed {
     pub field1: Option<u32>,
     pub field2: Option<u32>,
 }
 
-impl MyStruct {
+impl MyStructRenamed {
     pub fn new() -> Self {
         Default::default()
     }
 
-    pub fn builder() -> MyStructBuilder {
-        MyStructBuilder::new()
+    pub fn builder() -> MyStructRenamedBuilder {
+        MyStructRenamedBuilder::new()
     }
 }
 
-impl aldrin::private::aldrin_core::Serialize for MyStruct {
+impl aldrin::private::aldrin_core::Serialize for MyStructRenamed {
     fn serialize(&self, serializer: aldrin::private::aldrin_core::Serializer) -> Result<(), aldrin::private::aldrin_core::SerializeError> {
         let mut serializer = serializer.serialize_struct(2)?;
 
@@ -34,7 +34,7 @@
     }
 }
 
-impl aldrin::private::aldrin_core::Deserialize for MyStruct {
+impl aldrin::private::aldrin_core::Deserialize for MyStructRenamed {
     fn deserialize(deserializer: aldrin::private::aldrin_core::Deserializer) -> Result<Self, aldrin::private::aldrin_core::DeserializeError> {
         let mut deserializer = deserializer.deserialize_struct()?;
 
@@ -59,7 +59,7 @@
 }
 
 #[derive(Debug, Clone, Default)]
-pub struct MyStructBuilder {
+pub struct MyStructRenamedBuilder {
     #[doc(hidden)]
     field1: Option<u32>,
 
@@ -68,7 +68,7 @@
 
 }
 
-impl MyStructBuilder {
+impl MyStructRenamedBuilder {
     pub fn new() -> Self {
         Default::default()
     }
@@ -83,8 +83,8 @@
         self
     }
 
-    pub fn build(self) -> MyStruct {
-        MyStruct {
+    pub fn build(self) -> MyStructRenamed {
+        MyStructRenamed {
             field1: self.field1,
             field2: self.field2,
         }
