--- example1-orig.rs
+++ example1-patched.rs
@@ -8,22 +8,22 @@
 
 #[derive(Debug, Clone, Default)]
 #[non_exhaustive]
-pub struct MyStructRenamed {
+pub struct MyStructRenamedAgain {
     pub field1: Option<u32>,
     pub field2: Option<u32>,
 }
 
-impl MyStructRenamed {
+impl MyStructRenamedAgain {
     pub fn new() -> Self {
         Default::default()
     }
 
-    pub fn builder() -> MyStructRenamedBuilder {
-        MyStructRenamedBuilder::new()
+    pub fn builder() -> MyStructRenamedAgainBuilder {
+        MyStructRenamedAgainBuilder::new()
     }
 }
 
-impl aldrin_client::private::aldrin_proto::Serialize for MyStructRenamed {
+impl aldrin_client::private::aldrin_proto::Serialize for MyStructRenamedAgain {
     fn serialize(&self, serializer: aldrin_client::private::aldrin_proto::Serializer) -> Result<(), aldrin_client::private::aldrin_proto::SerializeError> {
         let mut serializer = serializer.serialize_struct(2)?;
 
@@ -34,7 +34,7 @@
     }
 }
 
-impl aldrin_client::private::aldrin_proto::Deserialize for MyStructRenamed {
+impl aldrin_client::private::aldrin_proto::Deserialize for MyStructRenamedAgain {
     fn deserialize(deserializer: aldrin_client::private::aldrin_proto::Deserializer) -> Result<Self, aldrin_client::private::aldrin_proto::DeserializeError> {
         let mut deserializer = deserializer.deserialize_struct()?;
 
@@ -59,7 +59,7 @@
 }
 
 #[derive(Debug, Clone, Default)]
-pub struct MyStructRenamedBuilder {
+pub struct MyStructRenamedAgainBuilder {
     #[doc(hidden)]
     field1: Option<u32>,
 
@@ -68,7 +68,7 @@
 
 }
 
-impl MyStructRenamedBuilder {
+impl MyStructRenamedAgainBuilder {
     pub fn new() -> Self {
         Default::default()
     }
@@ -83,8 +83,8 @@
         self
     }
 
-    pub fn build(self) -> MyStructRenamed {
-        MyStructRenamed {
+    pub fn build(self) -> MyStructRenamedAgain {
+        MyStructRenamedAgain {
             field1: self.field1,
             field2: self.field2,
         }
