--- example1-orig.rs
+++ example1-patched.rs
@@ -1,7 +1,7 @@
 #[derive(::std::fmt::Debug, ::std::clone::Clone, ::std::default::Default, ::aldrin::Serialize, ::aldrin::Deserialize, ::aldrin::AsSerializeArg)]
 #[aldrin(crate = "::aldrin::core", schema = "example1")]
 #[non_exhaustive]
-pub struct r#MyStruct {
+pub struct r#MyStructRenamed {
     #[aldrin(id = 1, optional)]
     pub r#field1: ::std::option::Option<::std::primitive::u32>,
 
@@ -9,18 +9,18 @@
     pub r#field2: ::std::option::Option<::std::primitive::u32>,
 }
 
-impl r#MyStruct {
+impl r#MyStructRenamed {
     pub fn new() -> Self {
         <Self as ::std::default::Default>::default()
     }
 
-    pub fn builder() -> r#MyStructBuilder {
-        r#MyStructBuilder::new()
+    pub fn builder() -> r#MyStructRenamedBuilder {
+        r#MyStructRenamedBuilder::new()
     }
 }
 
 #[derive(::std::fmt::Debug, ::std::clone::Clone, ::std::default::Default)]
-pub struct r#MyStructBuilder {
+pub struct r#MyStructRenamedBuilder {
     #[doc(hidden)]
     r#field1: ::std::option::Option<::std::primitive::u32>,
 
@@ -29,7 +29,7 @@
 
 }
 
-impl r#MyStructBuilder {
+impl r#MyStructRenamedBuilder {
     pub fn new() -> Self {
         <Self as ::std::default::Default>::default()
     }
@@ -44,8 +44,8 @@
         self
     }
 
-    pub fn build(self) -> r#MyStruct {
-        r#MyStruct {
+    pub fn build(self) -> r#MyStructRenamed {
+        r#MyStructRenamed {
             r#field1: self.r#field1,
             r#field2: self.r#field2,
         }
