--- example1-orig.rs
+++ example1-patched.rs
@@ -9,7 +9,7 @@
 #[derive(Debug, Clone, Default, aldrin_client::private::aldrin_proto::Serialize, aldrin_client::private::aldrin_proto::Deserialize)]
 #[aldrin(crate = "aldrin_client::private::aldrin_proto")]
 #[non_exhaustive]
-pub struct MyStructRenamed {
+pub struct MyStructRenamedAgain {
     #[aldrin(id = 1)]
     pub field1: Option<u32>,
 
@@ -18,18 +18,18 @@
 
 }
 
-impl MyStructRenamed {
+impl MyStructRenamedAgain {
     pub fn new() -> Self {
         Default::default()
     }
 
-    pub fn builder() -> MyStructRenamedBuilder {
-        MyStructRenamedBuilder::new()
+    pub fn builder() -> MyStructRenamedAgainBuilder {
+        MyStructRenamedAgainBuilder::new()
     }
 }
 
 #[derive(Debug, Clone, Default)]
-pub struct MyStructRenamedBuilder {
+pub struct MyStructRenamedAgainBuilder {
     #[doc(hidden)]
     field1: Option<u32>,
 
@@ -38,7 +38,7 @@
 
 }
 
-impl MyStructRenamedBuilder {
+impl MyStructRenamedAgainBuilder {
     pub fn new() -> Self {
         Default::default()
     }
@@ -53,8 +53,8 @@
         self
     }
 
-    pub fn build(self) -> MyStructRenamed {
-        MyStructRenamed {
+    pub fn build(self) -> MyStructRenamedAgain {
+        MyStructRenamedAgain {
             field1: self.field1,
             field2: self.field2,
         }
